### DISCUSSION TITLE
What's the rationale of passing `body` to Jinja templates?

### DISCUSSION THREAD

**Discussion by floodpants** [sabuhish/fastapi-mail#79] [Q&A] [answered]
Is there a specific reason behind using `template.render(body=template_body)` when rendering J2 templates, as opposed to the more flexible `template.render(**template_body)`? Using the latter would allow much greater flexibility when sharing templates/base templates/includes between mail and other use cases within an application.

**Comment by maestro-1** [comment, marked as answer]
Good observation. Essentially it was to give more structure to how different parts of the fastapi-mail interacted at the time, but I have since observed that it reduces flexibility in the use of jinja2 in fastapi-mail. This is something I started working to update but have not gotten around to finishing and submitting a PR on because of other things currently occupying my time. If you have PR already that would be fantastic and I'd be happy to help review it as well

**Reply by floodpants** [reply]
PR #80 created

**Reply by sabuhish** [reply]
Sorry guys I have been quite busy with tasks, could check this. I have been looking for this really, indeed this much less code, makes sense move away from the body field. Thanks for your PR. I will try to deploy it today. Awesome work! Thanks a lot!

**Reply by jdvalentine** [reply]
This is a very good change and I see the logic of it - thanks everyone. I just want to add that, having implemented the legacy behaviour, my code broke with an error error after the update. I do appreciate that there will be breaking changes and once I changed from html to template_body in the MessageSchema object and removed the body references in the Jinja2 template, everything worked again but I think a good error message to explain that template_body should be provided if template_name= is set during the send, it would be good.

```File "app/helpers/emailhelper.py", line 187, in send_contact_us_email
    await self.send_mail(email)
  File "app/helpers/emailhelper.py", line 26, in send_mail
    await fm.send_message(message, template_name="base.html")
  File "lib/python3.9/site-packages/fastapi_mail/fastmail.py", line 113, in send_message
    msg = await self.__prepare_message(message, template)
  File "lib/python3.9/site-packages/fastapi_mail/fastmail.py", line 81, in __prepare_message
    template_data = self.make_dict(template_body)
  File "lib/python3.9/site-packages/fastapi_mail/fastmail.py", line 59, in make_dict
    return dict(data)
TypeError: 'NoneType' object is not iterable```

But all this said, this should have been handled by my tests. Thanks for a great project.

**Reply by sabuhish** [reply]
Sorry to hear that broke your code, I am happy that you already figured out the issue. Yes, I believe too this will make much more sense, we have check that checks conditions if it template name is given but does not check the template body you got a right to point. Thank you for your good observation and for your comments.
